os_dep/linux/os_intfs.c

```diff
@@ -1578,7 +1578,11 @@ static int rtw_net_set_mac_address(struct net_device *pnetdev, void *addr)
 	}
 
 	_rtw_memcpy(adapter_mac_addr(padapter), sa->sa_data, ETH_ALEN); /* set mac addr to adapter */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 15, 0))
+	eth_hw_addr_set(pnetdev, sa->sa_data); /* set mac addr to net_device */
```

Comment:
This helper [was first introduced in 5.15](https://github.com/torvalds/linux/commit/48eab831ae8b9f7002a533fa4235eed63ea1f1a3). A similar change was made upstream to upstream staging rtl drivers [here](https://github.com/torvalds/linux/commit/6ed178cb23ec7503dd354c02c0c9ef97a6b8b22a).